driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -292,6 +300,117 @@ static GeoNearSearchOperator near(final Point origin, final Number pivot, final
                 .append("pivot", notNull("pivot", pivot)));
     }
 
+    /**
+     * Returns a {@link SearchOperator} that searches for documents where the
+     * value or array of values at a given path contains any of the specified values
+     *
+     * @param path The indexed field to be searched.
+     * @param value The boolean value to search for.
+     * @param values More fields to be searched.
+     * @return The requested {@link SearchOperator}.
+     * @mongodb.atlas.manual atlas-search/in/ in operator
+     */
+    static InSearchOperator in(final FieldSearchPath path, final boolean value, final boolean... values) {
+        Assertions.notNull("values", values);
+        List<BsonValue> list = new ArrayList<>();
+        list.add(new BsonBoolean(value));
+        for (boolean n : values) {
+            list.add(new BsonBoolean(n));
+        }
+        BsonArray bsonArray = new BsonArray(list);
+        return in(notNull("path", path), bsonArray);
```

Comment:
- Despite the API not allowing arrays to be `null`, we don't enforce that with runtime checks in other places of the search API (obviously, a mistake). If we want to start doing that, we should do that everywhere. Or, we may keep not enforcing that.
- `Assertions.notNull` is called as just `notNull` everywhere else in `SearchOperator.java`.
- There is no need for the intermediate `List<BsonValue>`.
- `n` is a surprising variable name in this case. Let's use `v`, for example?

```suggestion
        BsonArray bsonArray = new BsonArray();
        bsonArray.add(new BsonBoolean(value));
        for (boolean v : values) {
            bsonArray.add(new BsonBoolean(v));
        }
        return in(notNull("path", path), bsonArray);
```

Reply:
1. Since it is not a potential design decision, but a bug (I say this because I agree it is an obvious mistake), to not nullcheck varargs, I think we should not remove the fix from this method, just because it is present everywhere else. I will leave it, and I have added a comment to the tracking ticket.
2. ok
3. I think the primary reason for this is performance, which I'm not sure matters much here. But, I have changed it here and elsewhere in the codebase for consistency.
4. ok

Reply:
> I think the primary reason for this is performance, which I'm not sure matters much here.

That's not the objection. There was unnecessary code that improved neither readability, nor performance, nor anything else as far as I can tell. We should not have such code.

I don't know how you found that pattern also in ``BsonArrayCodec.java`, but its cool that you did 👍